docs/tutorials/forum-application/6.main.md

```diff
@@ -0,0 +1,200 @@
+# Main function
+
+**This main function is responsible for running the Forum Application blockchain.**
+
+```go
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	app "github.com/andynog/abci2-forum-app/abci"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"syscall"
+
+	"github.com/cometbft/cometbft/p2p"
+	"github.com/cometbft/cometbft/proxy"
+	"github.com/spf13/viper"
+
+	db "github.com/cometbft/cometbft-db"
+	cfg "github.com/cometbft/cometbft/config"
+	cmtflags "github.com/cometbft/cometbft/libs/cli/flags"
+	cmtlog "github.com/cometbft/cometbft/libs/log"
+	nm "github.com/cometbft/cometbft/node"
+	"github.com/cometbft/cometbft/privval"
+)
+
+var homeDir string
+
+func init() {
+	flag.StringVar(&homeDir, "cmt-home", "", "Path to the CometBFT config directory (if empty, uses $HOME/.cometbft)")
+}
+
+func main() {
+	flag.Parse()
+	if homeDir == "" {
+		homeDir = os.ExpandEnv("$HOME/.cometbft")
+	}
+
+	config := cfg.DefaultConfig()
```

Comment:
What if we already have a config file? (e.g., we are recovering)

Reply:
the config parameters are still considered in the line `viper.SetConfigFile(fmt.Sprintf("%s/%s", homeDir, "config/config.toml"))`, the app will use the `config.toml` values